src/irreps/anirrep.jl

```diff
@@ -0,0 +1,304 @@
+    N < 4 && return 1 # abelian
+    a, b = x.a, x.b
+    if a == b
+        return a == 3 ? 5 : 1 # 3 x 3 = 1 + 1' + 1'' + 2*3
+    else
+        return 1
+    end
+end
+
+# TODO: any way to shorten this?
+function Base.iterate(p::ANIrrepProdIterator{N}, state::Int = 1) where {N}
+    a, b = p.a, p.b
+    if N < 4
+        return state > 1 ? nothing : (ANIrrep{N}((a.n + b.n) % _npartitions(N)), state + 1)
+    else
+        u, u′, u′′, three = ANIrrep{4}(0), ANIrrep{4}(1), ANIrrep{4}(2), ANIrrep{4}(3)
+        if state == 1
+            # rules with the trivial irrep
+            a.n == 0 && return (b, 2)
+            b.n == 0 && return (a, 2)
+            # 1' rules
+            if a.n + b.n == 3 # (1, 2) or (2, 1)
+                return (u, 2)
+            elseif a.n == b.n != 3 # (1,1) or (2,2)
+                return (A4Irrep((2 * a.n) % 3), 2)
+            elseif (a.n in (1, 2) && b.n == 3) || (a.n == 3 && b.n in (1, 2))
+                # all 1D × 3 = 3
+                return (three, 2)
+            else # 3 x 3
+                return (u, 2) # first in sequence
+            end
+        elseif state == 2 && a.n == 3 && b.n == 3
+            return (u′, 3)
+        elseif state == 3 && a.n == 3 && b.n == 3
+            return (u′′, 4)
+        elseif state == 4 && a.n == 3 && b.n == 3
+            return (three, 5)
+        else
+            return nothing
+        end
+    end
+end
+
+# Topological data
+# ----------------
+function dim(a::ANIrrep{N}) where {N}
+    N < 4 && return 1
+    return a.n == 3 ? 3 : 1
+end
+
+function Nsymbol(a::ANIrrep{N}, b::ANIrrep{N}, c::ANIrrep{N}) where {N}
+    N < 4 && return (c.n == (a.n + b.n) % _npartitions(N)) ? 1 : 0
+    # N = 4
+    u, u′, u′′, three = ANIrrep{4}(0), ANIrrep{4}(1), ANIrrep{4}(2), ANIrrep{4}(3)
+    if a == three
+        if b == three # 3 x 3
+            return (c == u || c == u′ || c == u′′) ? 1 : 2
+        else # 3 x 1D
+            return c == three ? 1 : 0
+        end
+    else
+        if b == three # 1D x 3
+            return c == three ? 1 : 0
+        else # 1D x 1D
+            return c.n == (a.n + b.n) % 3 ? 1 : 0
+        end
+    end
+end
+
+function Fsymbol(a::I, b::I, c::I, d::I, e::I, f::I) where {N, I <: ANIrrep{N}}
+    T = sectorscalartype(I)
+    Nabe = Nsymbol(a, b, e)
+    Necd = Nsymbol(e, c, d)
+    Nbcf = Nsymbol(b, c, f)
+    Nafd = Nsymbol(a, f, d)
+
+    Nabe > 0 && Necd > 0 && Nbcf > 0 && Nafd > 0 ||
+        return zeros(sectorscalartype(I), Nabe, Necd, Nbcf, Nafd)
+
+    # fallback through fusiontensor
+    A = fusiontensor(a, b, e)
+    B = fusiontensor(e, c, d)[:, :, 1, :]
+    C = fusiontensor(b, c, f)
+    D = fusiontensor(a, f, d)[:, :, 1, :]
+    @tensor F[-1, -2, -3, -4] := conj(D[1, 5, -4]) * conj(C[2, 4, 5, -3]) *
+        A[1, 2, 3, -1] * B[3, 4, -2]
+
+    return F
+end
+
+# bosonic
+function Rsymbol(a::I, b::I, c::I) where {N, I <: ANIrrep{N}}
+    R = convert(sectorscalartype(I), Nsymbol(a, b, c))
```

Comment:
This doesn't seem to match the data https://github.com/QuantumKitHub/CategoryData.jl/blob/data/data/Rsymbols/RepA4.txt

(bosonic sectors can still have non-trivial Rsymbols, e.g. SU2)